validator/src/journal/chain.rs

```diff
@@ -531,6 +531,63 @@ impl<
         }
     }
 
+    // Returns all blocks in forks not on the chain with the given head. If head is None, uses the
+    // current chain head. If head is not found, returns None.
+    pub fn forks(&self, head: &str) -> Option<Vec<Block>> {
+        let state = self
+            .state
+            .read()
+            .expect("No lock holder should have poisoned the lock");
+
+        match state.block_manager.ref_block(head) {
+            Err(BlockManagerError::UnknownBlock) => {
+                return None;
+            }
+            Err(err) => {
+                error!("Unexpected error occurred: {:?}", err);
```

Comment:
UnknownBlock is the only error variant that can be returned from ref_block(), but since we have not properly defined which methods return which errors, this is the only thing that makes sense to do.

For the BlockManager, we have one error type `BlockManagerError` for all methods. IMO, a better pattern is to define a new error type for each method so that you can be more specific about what the errors are for each method. As an example `ref_block()` would have:

```
pub enum BlockManagerRefError {
  UnknownBlock,
}
```